### PR TITLE
add mattermost

### PR DESCRIPTION
**Pull request by goern**
This is a container image to run mattermost, it required a mysql container.

### CODE REVIEW

File: mattermost/README.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+Review the file `config.json` carefully and adopt it to your needs, it will be
+put inside of the container image.
+
+### Building
+
+Just a simple `docker build --rm --tag <username>/mattermost .` will do it.
+
+## Running
+
+Start a MySQL container first, mattermost will need it: `docker run -ti --name mysql -e MYSQL_ALLOW_EMPTY_PASSWORD=true -e MYSQL_USER=mmuser -e MYSQL_PASSWORD=mostest -e MYSQL_DATABASE=mattermost_test mariadb` This will use a mariadb from hub.docker.io. user, password and database must
```
Comment (scollier) [inline mattermost/README.md]: What version of MySQL is required here?

File: mattermost/mattermost-launch.sh [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+  cp -a /opt/mattermost/config/config.json /opt/mattermost/config/config.json.orig
+fi
+
+if [ "$DRIVER_NAME" = "mysql" ]; then
+    sed -e 's#"DriverName": "mysql"#"DriverName": "'"$DRIVER_NAME"'"#' \
+        -e 's#"DataSource": "mmuser:mostest@tcp(mysql:3306)/mattermost_test?charset=utf8mb4,utf8"#"DataSource": "'"$MYSQL_USER:$MYSQL_PASSWORD@tcp($(printenv ${DATABASE_SERVICE_NAME^^}_SERVICE_HOST):$(printenv ${DATABASE_SERVICE_NAME^^}_SERVICE_PORT))/$MYSQL_DATABASE?charset=utf8mb4,utf8"'"#' \
+        /opt/mattermost/config/config.json.orig > /opt/mattermost/config/config.json
+fi
+
+cd /opt/mattermost/bin/
```
Comment (scollier) [inline mattermost/mattermost-launch.sh]: Why cd here, but then exec from full relative path?  Should this be WORKDIR and then just exec the binary?

File: mattermost/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+LABEL io.projectatomic.nulecule.environment.required="MYSQL_USER, MYSQL_PASSWORD, MYSQL_DATABASE" \
+      io.projectatomic.nulecule.environment.optional="VOLUME_CAPACITY" \
+      io.projectatomic.nulecule.volume.data="/var/lib/psql/data,1Gi"
+
+RUN dnf update -y && \
+    dnf install -y tar --setopt=tsflags=nodocs && \
+    dnf clean all
+
+RUN cd /opt && \
+    curl -LO https://releases.mattermost.com/2.1.0/mattermost-team-2.1.0-linux-amd64.tar.gz && \
```
Comment (scollier) [inline mattermost/Dockerfile]: Do they have a link to "latest"? How will this be kept up to date once version bumps?

Comment (goern) [inline mattermost/Dockerfile]: rechkpls

Comment (scollier) [inline mattermost/Dockerfile]: @goern still some open questions.

**Comment by scollier** [comment]
@goern and I spoke about this, we'll stick to a certain version of mattermost here, noted in the tag and README.

**Comment by scollier** [comment]
thx @goern  LGTM.

/cc @jzb

**Pull request merged**